### PR TITLE
Add special DDR field formatter for pointers to J9ConstantPool

### PR DESCRIPTION
**Pull request by keithc-ca**
In some contexts, the low-order bits of pointers to `J9ConstantPool` are used as flags: they must be masked before suggesting a `!j9constantpool` command.

Also print the flags embedded in those pointers.

Use `ConstantPoolHelpers` more widely.

Fixes: #6887

### CODE REVIEW

**Comment by pshipton** [comment]
@gacholio fyi the change you wanted, if you want to do any reviewing.

File: debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/commands/LocalMapCommand.java [hunk 1]
```diff
@@ -78,7 +78,7 @@ public void run(String command, String[] args, Context context, PrintStream out)
 			UDATA offsetPC = new UDATA(pc.sub(U8Pointer.cast(localMethod.bytecodes())));
 			CommandUtils.dbgPrint(out, "Relative PC = %d\n", offsetPC.longValue());
 
-			J9ClassPointer localClass = J9_CLASS_FROM_CP(localMethod.constantPool());
+			J9ClassPointer localClass = ConstantPoolHelpers.J9_CP_FROM_METHOD(localMethod).ramClass();
```
Comment (pshipton) [inline debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/commands/LocalMapCommand.java]: You changed other places to use `ConstantPoolHelpers.J9_CLASS_FROM_METHOD()` but not here.

Comment (keithc-ca) [reply debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/commands/LocalMapCommand.java]: I missed that one; I'll fix it and any others I can find.

Comment (keithc-ca) [reply debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/commands/LocalMapCommand.java]: Done.

**Comment by pshipton** [comment]
jenkins test sanity,extended plinux jdk8

**Pull request merged**